src/components/ChannelPreview/hooks/useChannelPreviewDisplayName.js

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect, useContext } from 'react';
+import { ChatContext } from '../../../context';
+
+export const useChannelPreviewDisplayName = (channel) => {
+  const { client } = useContext(ChatContext);
+  const [displayName, setDisplayName] = useState('');
+
+  useEffect(() => {
+    if (channel.data.name && typeof channel.data.name === 'string') {
+      setDisplayName(channel.data.name);
+      return;
+    }
+
+    const members = channel.state ? Object.values(channel.state.members) : [];
```

Comment:
optional chaining. also just wondering but is `state.members` an object or an array? the name would suggest it's an array
```suggestion
    const members = Object.values(channel?.state?.members || {});
```

Reply:
it's an object with IDs as keys and channel member responses as values